x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/add_prebuilt_rules_table/add_prebuilt_rules_table_filters.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFilterGroup, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { isEqual } from 'lodash/fp';
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+import { ADD_PREBUILT_RULES_TABLE_ACTIONS } from '../../../../../common/lib/apm/user_actions';
+import { useStartTransaction } from '../../../../../common/lib/apm/use_start_transaction';
+import * as i18n from './translations';
+import { TagsFilterPopover } from '../rules_table_filters/tags_filter_popover';
+import { RuleSearchField } from '../rules_table_filters/rule_search_field';
+import { useAddPrebuiltRulesTableContext } from './add_prebuilt_rules_table_context';
+
+const FilterWrapper = styled(EuiFlexGroup)`
+  margin-bottom: ${({ theme }) => theme.eui.euiSizeM};
+`;
+
+/**
+ * Collection of filters for filtering data within the Add Prebuilt Rules table.
+ * Contains search bar and tag selection
+ */
+const AddPrebuiltRulesTableFiltersComponent = () => {
+  const { startTransaction } = useStartTransaction();
+  const {
+    state: { filterOptions, tags },
+    actions: { setFilterOptions },
+  } = useAddPrebuiltRulesTableContext();
+
+  const { tags: selectedTags } = filterOptions;
+
+  const handleOnSearch = useCallback(
+    (filterString) => {
+      startTransaction({ name: ADD_PREBUILT_RULES_TABLE_ACTIONS.FILTER });
+      setFilterOptions((filters) => ({
+        ...filters,
+        filter: filterString.trim(),
+      }));
+    },
+    [setFilterOptions, startTransaction]
+  );
+
+  const handleSelectedTags = useCallback(
+    (newTags: string[]) => {
+      if (!isEqual(newTags, selectedTags)) {
```

Comment:
I'm wondering whether this `isEqual` check is necessary. How frequently is the callback invoked with identical tags?